independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/InjectionPointModifier.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright 2019 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.quarkus.arc.processor;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Set;
+import org.jboss.jandex.AnnotationInstance;
+import org.jboss.jandex.AnnotationTarget;
+import org.jboss.jandex.Type;
+
+/**
+ * Holds all {@link InjectionPointsTransformer} instances and provides helper method to apply them on a given
+ * type. This is used in {@link InjectionPointInfo}, prior to creating actual injection points.
+ *
+ * In order to operate on up-to-date annotations, this class operates also leverages {@link AnnotationStore}.
+ *
+ * @author Matej Novotny
+ */
+public class InjectionPointModifier {
+
+    private List<InjectionPointsTransformer> transformers;
+    private BuildExtension.BuildContext buildContext;
+    private AnnotationStore annotationStore;
+
+    InjectionPointModifier(List<InjectionPointsTransformer> tranformers, BuildExtension.BuildContext buildContext) {
+        this.buildContext = buildContext;
+        this.transformers = tranformers;
+        this.annotationStore = buildContext != null ? buildContext.get(BuildExtension.Key.ANNOTATION_STORE) : null;
+    }
+
+    public Set<AnnotationInstance> applyTransformers(Type type, AnnotationTarget target, Set<AnnotationInstance> qualifiers) {
+        TransformationContextImpl transformationContext = new TransformationContextImpl(target, qualifiers,
```

Comment:
If no transformers are registered, this method should return the original set of annotations immediately.

Reply:
Very good point